### PR TITLE
proposal: Added proposal for vertical block sharding (limit size of block)

### PR DESCRIPTION
**Pull request by bwplotka**
Related to https://github.com/thanos-io/thanos/issues/1424 and https://github.com/thanos-io/thanos/issues/3068

Signed-off-by: Bartlomiej Plotka <bwplotka@gmail.com>

### CODE REVIEW

File: docs/proposals/202011_vertical_block_sharding.md [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+
+## Alternatives
+
+* Shard number stored in external label.
+
+## Future Work
+
+* Add [`Cuckoofilter`](https://github.com/seiflotfy/cuckoofilter) to improve query performance on store GW as well with vertically sharded blocks.
+
+![Why Cuckoo filter: https://bdupras.github.io/filter-tutorial/](../img/cuckoo-pros.png)
```
Comment (cben) [inline docs/proposals/202011_vertical_block_sharding.md:106]: This image is missing?  https://bdupras.github.io/filter-tutorial/ link works :+1:

**Comment by bwplotka** [comment]
Very relevant (:

Update: We are planning to propose this issue for LFX

File: docs/proposals/202011_vertical_block_sharding.md [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+* Redesign Block index format 
+* Introduce breaking changes
+* Lift current limitation of index size
+* Automatically shard old blocks
+
+## Proposal
+
+**We propose to vertically shard all blocks which goes beyond the specified index size limit during the compaction process.**
+
+* Special metadata in meta.json indicating a shard.
```
Comment (bwplotka) [inline docs/proposals/202011_vertical_block_sharding.md:92]: * One problem to solve, is how to perform compaction from X sources that output in Y (where Y > 1) blocks.
* How to ensure multiple compactors can distribute compaction without strong coordination.
* How to ensure multiple store gateways can exclude certain blocks from lookup process without directly touching them. (aka Bloom/Cuckoo filter)

**Review by bwplotka — COMMENTED** [review]
Some thoughts in our 1:1 with @Biswajitghosh98

**Comment by bwplotka** [comment]
Actually I think @Biswajitghosh98 had opened counterpart

**Comment by bwplotka** [comment]
Relevant, but far in progress, unfortunately.

**Comment by bwplotka** [comment]
Newer version https://github.com/thanos-io/thanos/pull/4191

**Pull request closed without merging**